server/server.go

```diff
@@ -302,9 +304,37 @@ func NewServer(cfg *config.Config, driver IDriver) (*Server, error) {
 
 	variable.RegisterStatistics(s)
 
+	variable.UpdateTLSConfig = s.updateTLSConfig
+
 	return s, nil
 }
 
+func (s *Server) updateTLSConfig(oldDirPath string) error {
+	oldCert, oldKey := filepath.Join(oldDirPath, "cert.pem"), filepath.Join(oldDirPath, "key.pem")
+	if _, err := os.Stat(oldCert); os.IsNotExist(err) {
+		return nil
+	}
+	if _, err := os.Stat(oldKey); os.IsNotExist(err) {
+		return nil
+	}
+	if key, cert := variable.GetSysVar("ssl_key").Value, variable.GetSysVar("ssl_cert").Value; cert == oldCert || key == oldKey {
+		return nil
+	}
```

Comment:
Only if the paths of key and cert are not specified by configs, and there are already key and cert files in the temporary directory, the changing of `tidb_tmp_storage_path` would reload TLS.